features/upsert.feature

```diff
@@ -0,0 +1,43 @@
+Feature: Upsert Cache Time
+
+  Scenario: Create Cache Time resource
+    Given the document with "_id" set to "5d41402abc4b2a76b9719d911017c592" does not exist in the "cachetimes" collection
+    When I PUT "/v1/cache-times/5d41402abc4b2a76b9719d911017c592"
+            """
+            {
+                "path": "/my-path",
+                "etag": "test-etag",
+                "collection_id": 123456,
+                "release_time": "2024-01-31T01:23:45.678Z"
+            }
+            """
+    Then the HTTP status code should be "204"
+
```

Comment:
And a document should now exist?

Reply:
The reason why this wasn't done is that after one of our chats with James, it was suggested to use component testing as a way of doing black-box testing, so we're only making assertions on things that the end user can see (just the HTTP response, basically). The `mongo_feature.go` file in `dp-component-test`, which contains some MongoDB steps, seems to have been done following this idea, as it does not give you any steps for checking the status of MongoDB, instead you have steps for setting it up. It should be possible to write steps for checking MongoDB, but that would go against this idea of black-box testing.

In any case, all the tests I've added for this PR are already being covered by unit tests. The main purpose of this PR was to enable component testing again, so that the team can start using it. I think we need a ticket to refactor our tests, because a good amount of them could be rewritten as component tests.

Reply:
Sure.